mm2src/coins/lp_coins.rs

```diff
@@ -1895,6 +1896,12 @@ pub trait MakerCoinSwapOpsV2: ParseCoinAssocTypes + CommonSwapOpsV2 + Send + Syn
     async fn send_maker_payment_v2(&self, args: SendMakerPaymentArgs<'_, Self>) -> Result<Self::Tx, TransactionErr>;
 
     /// Validate maker payment transaction
+    ///
+    /// Important:
+    /// - Offline semantic validation only (destination address/script, value/ABI args/pubs).
+    /// - Must not perform any network I/O: no RPC calls, no mempool lookups, no confirmation checks.
+    /// - Presence and confirmations are enforced by the swap state machine
+    /// (e.g., before the taker waits for maker payment confirmation to allow for fast failure).
     async fn validate_maker_payment_v2(&self, args: ValidateMakerPaymentArgs<'_, Self>) -> ValidatePaymentResult<()>;
```

Comment:
Perhaps, add a `offline` prefix to the function (e.g., `offline_maker_payment_validation_v2`)?

Reply:
Well, validation in the crypto sense is an offline operation after fetching or getting all the needed info from the chain / p2p / light nodes / etc..., I think it would be excessive to add offline to this but it's fine by me if you insist.